### ISSUE TITLE
Implement proper error responses according to RFC

### ISSUE THREAD

**Issue by hjacobs** [zalando/planb-tokeninfo#11]
See https://tools.ietf.org/html/rfc6750#section-6.2

Example missing token response:
```
GET /oauth2/tokeninfo HTTP/1.1

HTTP/1.1 400 Bad Request
Content-Type: application/json;charset=UTF-8

{"error":"invalid_request","error_description":"Access Token not valid"}
```

**Comment by hjacobs** [comment]
Example invalid token response:

```
GET /oauth2/tokeninfo?access_token=foo HTTP/1.1

HTTP/1.1 400 Bad Request
Content-Type: application/json;charset=UTF-8

{"error":"invalid_request","error_description":"Access Token not valid"}
```

(These are example responses from our current Zalando OAuth 2 provider)

**Comment by hjacobs** [comment]
We should consider doing it like https://tools.ietf.org/html/rfc6750#section-3.1 says:

* "invalid_request" with status code 400 for missing parameters
* "invalid_token" with status code 401 for invalid or expires tokens

This differs from what our current OAuth 2 provider returns :disappointed:

**Comment by hjacobs** [comment]
Also the RFC mentions the `WWW-Authenticate` HTTP header...

> And in response to a protected resource request with an
>  authentication attempt using an expired access token:

```
HTTP/1.1 401 Unauthorized
WWW-Authenticate: Bearer realm="example",
                   error="invalid_token",
                   error_description="The access token expired"

**Comment by hjacobs** [comment]
We should also have a look at how the Google Token Info endpoint handles errors:

https://www.googleapis.com/oauth2/v3/tokeninfo

**Comment by hjacobs** [comment]
See https://developers.google.com/identity/protocols/OAuth2UserAgent

> If the token has expired, has been tampered with, or the permissions revoked, the Google Authorization Server will respond with an error.
> The error surfaces as a 400 status code, and a JSON body as follows:
>
> {"error":"invalid_token"}
> By design, no additional information is given as to the reason for the failure.

**Comment by lmineiro** [comment]
Can you please check if https://github.com/zalando/planb-tokeninfo/pull/9 already does some of this?

**Comment by hjacobs** [comment]
@lmineiro let's discuss this on Monday, I'm also not completely sure which way to go exactly --- but we should do it "right" :smirk:

**Comment by lukasniemeier-zalando** [comment]
As @hjacobs points out our current provider does not return 401 on _invalid_token_, it responds with 400. I've tried to get a patch merged on that (it really looks like a mistake), but patch was rejected because it would break compatibility with too many clients.

**Comment by hjacobs** [comment]
Proper RFC error codes will be implemented with #32 (https://tools.ietf.org/html/rfc7662#section-2.3) which also changes the behavior as invalid tokens will not cause a 401, but instead return {"active": false}
